### ISSUE TITLE
Explore failing //lte/gateway/c/core/oai/test/amf:amf_procedures_test with --config=asan

### ISSUE THREAD

**Issue by electronjoe** [magma/magma#12144]
This bazel test target is failing.

## Reproduction

- If you have access to GitHub Codespaces (you should because you are a member of GH Magma)
- Go over to my [Bazel prototype branch](https://github.com/electronjoe/magma/tree/pr-giant-bazel-mme-rebase)
- Click on the Green colored `Code` drop down button
- Select the `Codespaces` toggle
- Select `New Codespace` and when asked select 16 core (why not!)
- This will spin up a GitHub codespace that contains my branch
- Once the codespace is up (vs code in browswer) in the terminal window type `bazel test //lte/gateway/c/core/oai/test/amf:amf_procedures_test --config=asan` and hit enter. This triggers a bazel build of all necessary components to run the amf_procedures_test, then runs the test

**Comment by electronjoe** [comment]
This is being addressed by #12099

**Comment by themarwhal** [comment]
On master 
```bash
# PASSES
bazel test //lte/gateway/c/core/oai/test/amf:amf_procedures_test --runs_per_test=100
# FAILED 11/111
bazel test //lte/gateway/c/core/oai/test/amf:amf_procedures_test --runs_per_test=100 --config=asan
```

The test is still flaky and even more flaky with ASAN enabled. From a quick glance, I think there are actual ASAN issues being reported.
